Fix IPSec abbr to IPsec (#3163)
antrea-io/antrea: CONTRIBUTING.md

   cannot be fixed automatically, an error message will be displayed so you can address the issue.
3. Follow [git commit](https://chris.beams.io/posts/git-commit/) guidelines.
4. Follow [logging](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/logging.md) guidelines.
5. Check below table for some general words and abbreviations you might use.

| Word/Phrase | Word/Abbr | Notes |
|---|---|---|
| IP Security | IPsec | refer to [rfc 6071](https://datatracker.ietf.org/doc/html/rfc6071) |

> | Namespace, Pod, Service... | Namespace, Pod, Service... | Any Kubernetes object in log/comment should start with upper case |

Reviewer: I don't know if a table really makes sense here since not everything is going to be about abbreviations (the second rule for example). Maybe we should have a separate section `Conventions for Writing Documentation` in this doc and simply use a bullet point list for now. You can add a link to the new section here.

Author: done, Could you help to take a look again? thanks.

   cannot be fixed automatically, an error message will be displayed so you can address the issue.
3. Follow [git commit](https://chris.beams.io/posts/git-commit/) guidelines.
4. Follow [logging](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/logging.md) guidelines.

> 5. please refer to [Conventions for Writing Documentation](#conventions-for-writing-document) for
> some general words and abbreviations you may need to pay attention.

Reviewer: ```suggestion
5. Please refer to [Conventions for Writing Documentation](#conventions-for-writing-document) for
spelling conventions when writing documentation or commenting code.
```

Author: done

documentation on [cherry-picks](docs/contributors/cherry-picks.md) for more
information.

### Conventions for Writing Documentation

> * Short name of `IP Security` should be `IPsec` referring to [rfc 6071](https://datatracker.ietf.org/doc/html/rfc6071).

Reviewer: ```suggestion
* Short name of `IP Security` should be `IPsec` as per [rfc 6071](https://datatracker.ietf.org/doc/html/rfc6071).
```

Author: done
